### ISSUE TITLE
Scandinavian letters in classname fails class-name-casing rule

### ISSUE THREAD

**Issue by trubens** [typescript-eslint/typescript-eslint#1039]
**What were you trying to do?**

The rule '@typescript-eslint/class-name-casing' is triggered when I try to use a Scandinavian letter  (æøå) in the class name.

```json
{
  "rules": {
    "@typescript-eslint/class-name-casing": ["error"]
  }
}
```

```ts
class ThisClassContainsTheScandinavianLetteræButFails {
}
```

**What did you expect to happen?**
It should accept the name

**What actually happened?**
Class 'ThisClassContainsTheScandinavianLetteræButFails' must be PascalCased.eslint(@typescript-eslint/class-name-casing)

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.3.2` |
| `@typescript-eslint/parser`             | `2.3.2` |
| `TypeScript`                            | `3.6.3` |
| `node`                                  | `10.16.0` |
| `npm`                                   | `6.10.1` |

**Comment by bradzacher** [comment]
Happy to accept a PR to add support for non-english letters.
It's not something that generally comes up in code, as unicode characters in code isn't often aren't well handled (if at all) by parsers and engines.
